src/en/quantumscans/src/eu/kanade/tachiyomi/extension/en/quantumscans/QuantumScans.kt

```diff
@@ -1,20 +1,150 @@
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        return searchMangaParse(response)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$apiUrl/api/query".toHttpUrl().newBuilder().apply { // 'query' instead of 'posts'
+            addQueryParameter("page", page.toString())
+            addQueryParameter("perPage", "18")
+            addQueryParameter("orderBy", "updatedAt")
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+        val scriptContent = document.select("script:containsData(self.__next_f.push)")
+            .joinToString("\n") { it.data() }
+
+        if (scriptContent.isBlank()) {
+            throw Exception("Unable to find chapter data scripts.")
+        }
+
+        val urlRegex = Regex("""(https?://[^"]+?\.(?:jpg|jpeg|png|webp))""")
+
+        val imageUrls = urlRegex.findAll(scriptContent)
+            .map { it.groupValues[1].replace("\\\\", "") }
+            .filter { it.contains("/series/") || it.contains("/uploads/") }
+            .distinct()
+            .toList()
+
+        if (imageUrls.isEmpty()) {
+            // Fallback for when the main regex fails for some reason
+            val fallbackRegex = Regex("""(https?://[^"]+?/upload/series/[^"]+\.(?:jpg|jpeg|png|webp))""")
+            val fallbackUrls = fallbackRegex.findAll(scriptContent)
+                .map { it.groupValues[1].replace("\\\\", "") }
+                .distinct()
+                .toList()
+
+            if (fallbackUrls.isEmpty()) {
+                throw Exception("Could not extract any image URLs.")
+            }
+            return fallbackUrls.sortedWith(
+                compareBy { url -> url.substringAfterLast('/').substringBeforeLast('.').toIntOrNull() ?: -1 },
+            ).mapIndexed { index, url -> Page(index, imageUrl = url) }
+        }
+
+        return imageUrls.sortedWith(
+            compareBy { url -> url.substringAfterLast('/').substringBeforeLast('.').toIntOrNull() ?: -1 },
+        ).mapIndexed { index, url -> Page(index, imageUrl = url) }
+    }
+
+    private var genresList: List<Pair<String, String>> = emptyList()
+    private var fetchGenresAttempts = 0
+
+    private fun fetchGenres() {
+        if (fetchGenresAttempts < 3 && genresList.isEmpty()) {
+            try {
+                val response = client.newCall(GET("$apiUrl/api/genres", headers)).execute()
+                genresList = response.parseAs<List<GenreDto>>()
+                    .map { Pair(it.name, it.id.toString()) }
+            } catch (e: Throwable) {
+            } finally {
+                fetchGenresAttempts++
+            }
+        }
+    }
+
+    override fun getFilterList(): FilterList {
+        if (genresList.isEmpty() && fetchGenresAttempts == 0) {
```

Comment:
You probably want `fetchGenresAttempts < 3` here and remove the check in `fetchGenres()`.